### DISCUSSION TITLE
Prometheus: Expose individual probe results

### DISCUSSION THREAD

**Discussion by theduke** [cloudprober/cloudprober#531] [Q&A]
Currently the Prometheus surfacer just provides aggregate data. 

It would be great if we could also get an option to provide a metric for each individual executed probe, which is purged once it was exported once.

**Comment by manugarg** [comment]
@theduke You can export gauge metrics instead of aggregates by adding the following option to the config:

```
export_as_gauge: true
```

(or `exportAsGauge: true` in YAML)

https://cloudprober.org/docs/config/surfacer/#cloudprober_surfacer_SurfacerDef

Note: if you run more than one probe every 10s, you'll still see results aggregated every 10s. You can change this behavior by setting `stats_export_interval` (per probe config option) to a lower value than the probe interval.

**Comment by manugarg** [comment]
@theduke Can you please confirm if the above comment answers your question?
